src/UKHO.S100PermitService.StubService/Stubs/ProductKeyServiceStub.cs

```diff
@@ -8,21 +9,74 @@ namespace UKHO.S100PermitService.StubService.Stubs
 {
     public class ProductKeyServiceStub : IStub
     {
-        private readonly ProductKeyServiceConfiguration _productKeyServiceConfiguration;
+        public const string CONTENTTYPE = "Content-Type";
+        public const string APPLICATIONTYPE = "application/json";
+        private readonly ProductKeyServiceConfiguration productKeyServiceConfiguration;
 
         public ProductKeyServiceStub(ProductKeyServiceConfiguration productKeyServiceConfiguration)
         {
-            _productKeyServiceConfiguration = productKeyServiceConfiguration;
+            this.productKeyServiceConfiguration = productKeyServiceConfiguration;
         }
 
         public void ConfigureStub(WireMockServer server)
         {
-            server
-                .Given(Request.Create().WithPath(_productKeyServiceConfiguration.Url).UsingGet())
-                .RespondWith(Response.Create()
-                    .WithStatusCode(HttpStatusCode.OK)
-                    .WithHeader("Content-Type", "application/json")
-                    .WithBody("{ \"message\": \"PKS API Stub response\" }"));
+            server //401
+             .Given(Request.Create()
+             .WithPath(productKeyServiceConfiguration.Url)
+             .UsingPost()
+             .WithHeader("Authorization", "Bearer ", MatchBehaviour.RejectOnMatch))
+             .RespondWith(Response.Create()
+                .WithStatusCode(HttpStatusCode.Unauthorized)
+                .WithHeader(CONTENTTYPE, APPLICATIONTYPE)
+                .WithBody(@"{ ""result"": ""token missing""}"));
+
+            server //404
+                  .Given(Request.Create()
+                  .WithPath(productKeyServiceConfiguration.Url)
+                  .UsingPost()
+                  .WithHeader("Authorization", "Bearer *", MatchBehaviour.AcceptOnMatch))
+                  .RespondWith(Response.Create()
+                        .WithStatusCode(HttpStatusCode.NotFound)
+                        .WithHeader(CONTENTTYPE, APPLICATIONTYPE)
+                        .WithBody("{\r\n  \"correlationId\": \"4c2dec2a-f44f-445d-837f-99fa72836533\",\r\n  \"errors\": [\r\n    {\r\n      \"source\": \"GetProductKey\",\r\n      \"description\": \"Key not found for Product and Edition \"\r\n    }\r\n  ]\r\n}\r\n"));
```

Comment:
Please define this JSON in a file as elsewhere

Reply:
Moved